sdk/data/azcosmos/cosmos_container_read_many.go

```diff
@@ -6,34 +6,101 @@ package azcosmos
 import (
 	"context"
 	"errors"
-	"runtime"
 	"sync"
 
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
 	"github.com/Azure/azure-sdk-for-go/sdk/data/azcosmos/queryengine"
+	"github.com/Azure/azure-sdk-for-go/sdk/internal/log"
 )
 
 // executeReadManyWithEngine executes a query using the provided query engine.
 func (c *ContainerClient) executeReadManyWithEngine(queryEngine queryengine.QueryEngine, items []ItemIdentity, readManyOptions *ReadManyOptions, operationContext pipelineRequestOptions, ctx context.Context) (ReadManyItemsResponse, error) {
-	// throw error that this is unsupported
-	return ReadManyItemsResponse{}, errors.New("ReadMany with query engine is not supported yet")
+	path, _ := generatePathForNameBased(resourceTypeDocument, operationContext.resourceAddress, true)
+
+	// get the partition key ranges for the container
+	rawPartitionKeyRanges, err := c.getPartitionKeyRangesRaw(ctx, operationContext)
+	if err != nil {
+		// if we can't get the partition key ranges, return empty response
+		return ReadManyItemsResponse{}, err
+	}
+
+	// get the container properties
+	containerRsp, err := c.Read(ctx, nil)
+	if err != nil {
+		return ReadManyItemsResponse{}, err
+	}
+
+	// create the item identities for the query engine with json string
+	newItemIdentities := make([]queryengine.ItemIdentity, len(items))
+	for i := range items {
+		pkStr, err := items[i].PartitionKey.toJsonString()
+		if err != nil {
+			return ReadManyItemsResponse{}, err
+		}
+		newItemIdentities[i] = queryengine.ItemIdentity{
+			PartitionKeyValue: pkStr,
+			ID:                items[i].ID,
+		}
+	}
+	var pkVersion int32
+	if containerRsp.ContainerProperties.PartitionKeyDefinition.Version == 0 {
+		pkVersion = int32(1)
+	} else {
+		pkVersion = int32(containerRsp.ContainerProperties.PartitionKeyDefinition.Version)
+	}
+
+	readManyPipeline, err := queryEngine.CreateReadManyPipeline(string(rawPartitionKeyRanges), newItemIdentities, string(containerRsp.ContainerProperties.PartitionKeyDefinition.Kind), pkVersion)
```

Comment:
Since the client engine expects strings, we should serialize this into a JSON string before using it - also the ordering ideally changed
```suggestion
	newItemIdentitiesJSON, err := json.Marshal(newItemIdentities)
	if err != nil {
		return ReadManyItemsResponse{}, err
	}
	readManyPipeline, err := queryEngine.CreateReadManyPipeline(string(newItemIdentitiesJSON), string(rawPartitionKeyRanges), string(containerRsp.ContainerProperties.PartitionKeyDefinition.Kind), pkVersion)
```